src/ansys/grantami/bomanalytics/bom_types/_bom_reader.py

```diff
@@ -178,71 +176,59 @@ def _append_unserialized_fields(self, type_name: str, obj: Dict) -> None:
         obj: Dict
             The source dictionary with the data to be deserialized.
-        field_name: str
-            Local name of the target field.
-        namespace_url: Optional[str]
-            If the target namespace is different from that of the target object, for example if the type defines an
-            anonymous complex type, it can be overridden here.
+        field_name: QualifiedXMLName
+            Fully qualified name of the target field.
         """
-        if namespace_url is None:
-            namespace_url = instance.namespace
-
         for k, v in obj.items():
             if k.startswith("@xmlns"):
                 continue
 
-            if k == "$" and field_name == "$":
+            if k == "$" and field_name.local_name == "$":
                 del obj[k]
                 return v
 
             if k.startswith("@"):
-                is_matched = self._match_attribute(k, field_name, namespace_url)
+                is_matched = self._match_attribute(k, field_name)
                 if is_matched:
                     del obj[k]
                     return v
             else:
-                is_matched = self._match_element(k, field_name, namespace_url)
+                is_matched = self._match_element(k, field_name)
                 if is_matched:
                     del obj[k]
                     return v
         return None
 
-    def _match_element(self, item_name: str, field_name: str, namespace_url: str) -> bool:
+    def _match_element(self, item_name: str, match_name: QualifiedXMLName) -> bool:
         if ":" not in item_name:
-            return "" in self._namespaces and namespace_url == self._namespaces[""] and item_name == field_name
+            return match_name.namespace == self._namespaces.get("", None) and item_name == match_name.local_name
         namespace_prefix, stripped_name = item_name.split(":")
         field_namespace_url = self._namespaces[namespace_prefix]
-        return namespace_url == field_namespace_url and stripped_name == field_name
+        return match_name.namespace == field_namespace_url and stripped_name == match_name.local_name
 
-    def _match_attribute(self, item_name: str, field_name: str, namespace_url: str) -> bool:
+    def _match_attribute(self, item_name: str, match_name: QualifiedXMLName) -> bool:
         if not item_name.startswith("@"):
             return False
         if ":" not in item_name:
             if "" in self._namespaces:
-                return namespace_url == self._namespaces[""] and item_name == field_name
+                return match_name.namespace == self._namespaces[""] and item_name == match_name.local_name
             else:
                 # Workaround for https://github.com/ansys/grantami-bomanalytics-private/issues/75
-                # TODO - properly check the _parent_ object's namespace and make sure that we expect a namespace
```

Comment:
I think this implies that we foresaw this issue, or something similar to it. Either way, I no longer think it's relevant.